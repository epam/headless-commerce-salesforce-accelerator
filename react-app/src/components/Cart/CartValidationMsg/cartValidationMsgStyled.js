/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import styled from "styled-components";

export const StyledCartValidationError = styled.div`
  font: ${(props) => props.theme.font("regular", 12, 16)};
  display: flex;
  justify-content: center;
  margin: ${({ margin = 0 }) => margin};
  background-color: ${({ theme }) => theme.colors.pale};
  padding: 16px 16px 16px 0;
`;

export const StyledIcon = styled.div`
  margin: 0 11px 15px 15px;
`;
