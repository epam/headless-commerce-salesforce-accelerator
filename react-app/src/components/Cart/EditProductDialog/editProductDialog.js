/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import PropTypes from "prop-types";

/** UI Components */
import Grid from "@material-ui/core/Grid";

/** Components */
import Modal from "components/core/Modal";
import Carousel from "components/ProductDetails/Carousel";
import Description from "components/ProductDetails/Description";
import VariantSelectors from "components/ProductDetails/VariantSelectors";
import StyledDialogContent from "./editProductDialogStyled";

const EditProductDialog = ({ open, handleClose, updateProductInCart }) => (
  <Modal
    isOpen={open}
    handleClose={handleClose}
    modalTitle="Edit Item"
    submitAction={updateProductInCart}
    submitButtonText="Update"
  >
    <StyledDialogContent dividers>
      <Grid container spacing={3}>
        <Grid item xs={6}>
          <Carousel />
        </Grid>
        <Grid item xs={6}>
          <VariantSelectors />
          <Description
            titleFontSize={14}
            lineheight={24}
            titleMargin="24px 0px 8px 0px"
            contentMargin="0px 0px 24px 0px"
          />
        </Grid>
      </Grid>
    </StyledDialogContent>
  </Modal>
);

EditProductDialog.propTypes = {
  open: PropTypes.bool,
  handleClose: PropTypes.func,
  updateProductInCart: PropTypes.func,
};

export default EditProductDialog;
