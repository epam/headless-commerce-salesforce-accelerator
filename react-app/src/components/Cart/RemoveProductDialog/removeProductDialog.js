/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import { useSelector } from "react-redux";
import PropTypes from "prop-types";

/** Selectors */
import { selectCartResources } from "store/selectors/cart/selectors";

/** Components */
import Modal from "components/core/Modal";
import {
  StyledDialogContentText,
  StyledDialogContent,
} from "./removeProductStyled";

const RemoveProductDialog = ({
  open,
  handleClose,
  deleteProductFromCart,
  id,
  uuid,
}) => {
  const resources = useSelector(selectCartResources);

  const clickHandler = () => {
    deleteProductFromCart(id, uuid);
  };

  return (
    <Modal
      isOpen={open}
      handleClose={handleClose}
      modalTitle={resources?.cartRemoveProduct}
      submitAction={clickHandler}
      submitButtonText={resources?.remove}
    >
      <StyledDialogContent dividers>
        <StyledDialogContentText>
          {resources?.cartRemoveProductConfirmation}
        </StyledDialogContentText>
      </StyledDialogContent>
    </Modal>
  );
};

RemoveProductDialog.propTypes = {
  open: PropTypes.bool,
  handleClose: PropTypes.func,
  deleteProductFromCart: PropTypes.func,
  id: PropTypes.string,
  uuid: PropTypes.string,
};

export default RemoveProductDialog;
