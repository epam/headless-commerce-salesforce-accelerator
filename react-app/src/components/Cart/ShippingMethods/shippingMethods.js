/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import { useDispatch, useSelector } from "react-redux";
import Select from "components/core/Select";
import cartActions from "store/actions/cart/actions";

import {
  selectShipments,
  selectShipmentUUID,
} from "store/selectors/cart/selectors";

const ShippingMethods = () => {
  const dispatch = useDispatch();
  const shipments = useSelector(selectShipments);
  const shipmentUUID = useSelector(selectShipmentUUID);

  const handleChange = (e) => {
    dispatch(
      cartActions.selectShippingMethod(e.currentTarget.value, shipmentUUID)
    );
  };

  const setShippingMethodDisplayValue = (methods) => {
    return methods.map((method) => {
      const updatedMethod = method;
      const { displayName, estimatedArrivalTime } = updatedMethod;
      if (displayName && estimatedArrivalTime) {
        updatedMethod.displayValue = `${displayName} (${estimatedArrivalTime})`;
      } else if (displayName) {
        updatedMethod.displayValue = displayName;
      }

      return updatedMethod;
    });
  };

  const { shippingMethods, selectedShippingMethod } = shipments;

  return shipments && shipmentUUID ? (
    <Select
      values={setShippingMethodDisplayValue(shippingMethods)}
      defaultValue={selectedShippingMethod}
      label="Shipping"
      width={320}
      handleChange={handleChange}
      iconisrotated
      fullWidth
      margin="0 0 16px"
    />
  ) : null;
};

export default ShippingMethods;
