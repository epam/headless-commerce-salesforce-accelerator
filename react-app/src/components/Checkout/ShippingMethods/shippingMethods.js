/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useState, useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import RadioButtonIcon from "components/core/RadioButtonIcon";
import checkoutActions from "store/actions/checkout/actions";
import StyledTypography from "components/core/Typography";
import PropTypes from "prop-types";

/** UI Components */
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Radio from "@material-ui/core/Radio";
import RadioGroup from "@material-ui/core/RadioGroup";
import Grid from "@material-ui/core/Grid";

import {
  selectShipments,
  selectShipmentUUID,
  selectShippingMethodHeader,
  selectShipmentMethodFieldData,
} from "store/selectors/checkout/selectors";
import {
  StyledShippingMethods,
  StyledShippingItem,
} from "./shippingMethodsStyled";

const ShippingMethods = ({ register }) => {
  const dispatch = useDispatch();
  const shipments = useSelector(selectShipments);
  const shipmentUUID = useSelector(selectShipmentUUID);
  const shipmentMethodFieldData = useSelector(selectShipmentMethodFieldData);
  const shippingMethodHeader = useSelector(selectShippingMethodHeader);
  const { applicableShippingMethods } = shipments;
  const selectedShippingMethod = shipments.selectedShippingMethod.ID;
  const [value, setValue] = useState(selectedShippingMethod);

  useEffect(() => {
    setValue(selectedShippingMethod);
  }, [selectedShippingMethod]);

  const handleChange = (e) => {
    setValue(e.target.value);
    dispatch(
      checkoutActions.selectCheckoutShippingMethod(
        e.currentTarget.value,
        shipmentUUID
      )
    );
  };

  const renderRadioButton = (shippingMethod) => {
    const { displayName, estimatedArrivalTime, shippingCost } = shippingMethod;
    return (
      <StyledShippingItem key={shippingMethod.ID}>
        <FormControlLabel
          key={shippingMethod.ID}
          label={
            <Grid container>
              <Grid item xs={8}>
                <StyledShippingItem>
                  <StyledTypography display="inline">
                    {displayName}
                  </StyledTypography>
                  <StyledTypography
                    fontWeight="normal"
                    margin="0 0 0 5px"
                    display="inline"
                  >
                    ({estimatedArrivalTime})
                  </StyledTypography>
                </StyledShippingItem>
              </Grid>
              <Grid item xs={4}>
                <StyledTypography margin="0 0 0 20px" align="right">
                  {shippingCost}
                </StyledTypography>
              </Grid>
            </Grid>
          }
          value={shippingMethod.ID}
          id="shippingMethods"
          control={
            <Radio
              onChange={handleChange}
              icon={<RadioButtonIcon />}
              checkedIcon={<RadioButtonIcon checked />}
              name={shipmentMethodFieldData?.htmlName}
              inputRef={register()}
            />
          }
        />
      </StyledShippingItem>
    );
  };

  return shipments && shipmentUUID ? (
    <StyledShippingMethods>
      <StyledTypography
        variant="h3"
        fontSize={16}
        colortheme="mineShaft"
        margin="0 0 8px"
      >
        {shippingMethodHeader}
      </StyledTypography>
      <RadioGroup
        aria-label="shippingMethods"
        value={value}
        onChange={handleChange}
      >
        {applicableShippingMethods.map(renderRadioButton)}
      </RadioGroup>
    </StyledShippingMethods>
  ) : null;
};

ShippingMethods.propTypes = {
  register: PropTypes.func,
};

export default ShippingMethods;
