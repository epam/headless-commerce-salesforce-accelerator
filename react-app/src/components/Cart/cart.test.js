/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import PropTypes from "prop-types";

import CartValidationMsg from "./CartValidationMsg";
import EditProductDialog from "./EditProductDialog";
import RemoveProductDialog from "./RemoveProductDialog";
import ShippingMethods from "./ShippingMethods";

const ReduxProvider = ({ children, reduxStore }) => (
  <Provider store={reduxStore}>{children}</Provider>
);

ReduxProvider.propTypes = {
  children: PropTypes.element,
  reduxStore: PropTypes.shape({}),
};

describe("CartValidationMsg", () => {
  it("it should match snapshot", () => {
    const wrapper = shallow(
      <ReduxProvider>
        <CartValidationMsg />
      </ReduxProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe("EditProductDialog", () => {
  it("it should match snapshot", () => {
    const wrapper = shallow(<EditProductDialog />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("RemoveProductDialog", () => {
  it("it should match snapshot", () => {
    const wrapper = shallow(
      <ReduxProvider>
        <RemoveProductDialog />
      </ReduxProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe("ShippingMethods", () => {
  it("it should match snapshot", () => {
    const wrapper = shallow(
      <ReduxProvider>
        <ShippingMethods />
      </ReduxProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
