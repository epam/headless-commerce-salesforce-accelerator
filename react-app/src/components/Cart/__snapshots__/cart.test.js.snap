// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartValidationMsg it should match snapshot 1`] = `
<Provider>
  <CartValidationMsg />
</Provider>
`;

exports[`EditProductDialog it should match snapshot 1`] = `
<Modal
  modalTitle="Edit Item"
  submitButtonText="Update"
>
  <Styled(WithStyles(ForwardRef(DialogContent)))
    dividers={true}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <Carousel />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <VariantSelectors />
        <Description
          contentMargin="0px 0px 24px 0px"
          lineheight={24}
          titleFontSize={14}
          titleMargin="24px 0px 8px 0px"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </Styled(WithStyles(ForwardRef(DialogContent)))>
</Modal>
`;

exports[`RemoveProductDialog it should match snapshot 1`] = `
<Provider>
  <RemoveProductDialog />
</Provider>
`;

exports[`ShippingMethods it should match snapshot 1`] = `
<Provider>
  <ShippingMethods />
</Provider>
`;
